llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -2022,6 +2022,34 @@ void VPReductionEVLRecipe::execute(VPTransformState &State) {
   State.set(this, NewRed, /*IsScalar*/ true);
 }
 
+InstructionCost VPReductionRecipe::computeCost(ElementCount VF,
+                                               VPCostContext &Ctx) const {
+  RecurKind RdxKind = RdxDesc.getRecurrenceKind();
+  // TODO: Support any-of reduction and in-loop reduction
+  assert(!RecurrenceDescriptor::isAnyOfRecurrenceKind(RdxKind) &&
+         "Not support any-of reduction in VPlan-based cost model currently.");
+
```

Comment:
also assert that it's not an in-loop reduction?

Reply:
Add a new function `isInLoopReduction()` to check if the instruction is in the inLoopReduction chain.